2017-03-boston/lectures/bids-heudiconv/presentation.md

```diff
@@ -0,0 +1,413 @@
+    if (11 <= sl <= 22) and (nt == 1) and ('dti' in s[12]):
+      info[dwi].append(s[2]) # append if multiple scans meet criteria
+    if (nt > 10) and ('taskrest' in s[12]):
+      if s[13]: # motion corrected
+        info[rest].append({'item': s[2], 'acq': 'corrected'})
+      else:
+        info[rest].append({'item': s[2], 'acq': 'uncorrected'})
+```
+
+- Extract and label if resting state scans are motion corrected
+
+---
+### Our finished heuristic (`phantom_heuristic.py`)
+
+```python
+import os
+
+def create_key(template, outtype=('nii.gz',), annotation_classes=None):
+    if template is None or not template:
+        raise ValueError('Template must be a valid format string')
+    return template, outtype, annotation_classes
+
+def infotodict(seqinfo):
+
+    t1w = create_key('anat/sub-{subject}_T1w')
+    dwi = create_key('dwi/sub-{subject}_run-{item:01d}_dwi')
+    rest = create_key('func/sub-{subject}_task-rest_acq-{acq}_run-{item:01d}_bold')
+
+    info = {t1w: [], dwi: [], rest: []}
+
+    for s in seqinfo:
+        x, y, sl, nt = (s[6], s[7], s[8], s[9])
+        if (sl == 176) and (nt ==1) and ('t1' in s[12]):
+          info[t1w] = [s[2]] # assign if a single series meets criteria
+        if (11 <= sl <= 22) and (nt == 1) and ('dti' in s[12]):
+          info[dwi].append(s[2]) # append if multiple series meet criteria
+        if (nt > 10) and ('taskrest' in s[12]):
+            if s[13]: # exclude non motion corrected series
+                info[rest].append({'item': s[2], 'acq': 'corrected'})
+            else:
+                info[rest].append({'item': s[2], 'acq': 'uncorrected'})
+    return info
+```
+
+---
+### Changing our docker command
+
+.middle[
+```bash
+docker run --rm -it -v $PWD:/data nipy/heudiconv
+-d /data/%s/YAROSLAV_DBIC-TEST1/*/*/*IMA -s PHANTOM1_3
+-f /convertall.py -c none -o /data/output
+```
+]
+
+---
+### Changing our docker command
+
+.middle[
+```bash
+docker run --rm -it -v $PWD:/data nipy/heudiconv
+-d /data/%s/YAROSLAV_DBIC-TEST1/*/*/*IMA -s PHANTOM1_3
+-f /data/phantom_heuristic.py -c none -o /data/output
+```
+]
+
+---
+### Changing our docker command
+
+.middle[
+```bash
+docker run --rm -it -v $PWD:/data nipy/heudiconv
+-d /data/%s/YAROSLAV_DBIC-TEST1/*/*/*IMA -s PHANTOM1_3
+-f /data/phantom_heuristic.py -c dcm2niix -o /data/output
+```
+]
+
+---
+### Updated docker command
+
+.middle[
+```bash
+docker run --rm -it -v $PWD:/data nipy/heudiconv
+-d /data/%s/YAROSLAV_DBIC-TEST1/*/*/*IMA -s PHANTOM1_3
+-f /data/phantom_heuristic.py -c dcm2niix -b -o /data/output
+```
+]
+
+--
+
+- Clear old output directory
+- Run the docker command!
+- Something missing? Double check your `heuristic` and `dicominfo.txt`!
+
+---
+name: extrasteps
+
+### Is it BIDS yet?
+
+Let's check:
+- [In-browser validator](http://incf.github.io/bids-validator) / Command line
+
+--
+
+- A change to `heudiconv` will be coming soon that fixes most of these problems
```

Comment:
let's get these in. perhaps add a bids postprocessing function that gets called after conversion. whenever bids output is enabled. (in fact we should really consider refactoring this in the longer run with post-processors)